tests/snuba/test_snuba.py

```diff
@@ -142,3 +144,56 @@ def test_simple(self):
             {"issue": event_1.group.id, "event_id": event_1.event_id},
             {"issue": event_2.group.id, "event_id": event_2.event_id},
         ]
+
+    @mock.patch("sentry.utils.snuba._bulk_snuba_query", side_effect=snuba._bulk_snuba_query)
+    def test_cache(self, _bulk_snuba_query):
+        one_min_ago = iso_format(before_now(minutes=1))
+        event_1 = self.store_event(
+            data={"fingerprint": ["group-1"], "message": "hello", "timestamp": one_min_ago},
+            project_id=self.project.id,
+        )
+        event_2 = self.store_event(
+            data={"fingerprint": ["group-2"], "message": "hello", "timestamp": one_min_ago},
+            project_id=self.project.id,
+        )
+        params = [
+            snuba.SnubaQueryParams(
+                start=timezone.now() - timedelta(days=1),
+                end=timezone.now(),
+                selected_columns=["event_id", "group_id", "timestamp"],
+                filter_keys={"project_id": [self.project.id], "group_id": [event_1.group.id]},
+            ),
+            snuba.SnubaQueryParams(
+                start=timezone.now() - timedelta(days=1),
+                end=timezone.now(),
+                selected_columns=["event_id", "group_id", "timestamp"],
+                filter_keys={"project_id": [self.project.id], "group_id": [event_2.group.id]},
+            ),
+        ]
+
+        results = snuba.bulk_raw_query(
+            copy.deepcopy(params),
+            use_cache=True,
+        )
+        assert [
+            {"issue": r["data"][0]["group_id"], "event_id": r["data"][0]["event_id"]}
+            for r in results
+        ] == [
+            {"issue": event_1.group.id, "event_id": event_1.event_id},
+            {"issue": event_2.group.id, "event_id": event_2.event_id},
+        ]
+        assert _bulk_snuba_query.call_count == 1
+        _bulk_snuba_query.reset_mock()
+
```

Comment:
Could you maybe add a third event here and ensure that event is NOT returned in the cached query?

Reply:
Good thought, will do